Unity/Assets/Game/Audio/AudioManager.cs

```diff
@@ -0,0 +1,195 @@
+using System.Collections;
+using System.Collections.Generic;
+using JCMG.Utility;
+using NaughtyAttributes;
+using UnityEngine;
+using UnityEngine.Audio;
+
+public class AudioManager : Singleton<AudioManager>
+{
+	[SerializeField] private AudioSO _audioSettings;	//TODO add [SerializeField] private float _crossfadeTime;
+	[SerializeField] private AudioMixer _masterMixer;
+	[SerializeField] private AudioMixerGroup _sfxMixer, _bgmMixer;
+
+	private Dictionary<string, AudioSource> _audiosources;
+	private string _currentMusic = "";
+	private bool _crossfading = false;
+
+	protected override void Awake()
+	{
+		base.Awake();
+		_audiosources = new();
+	}
+
+	public void Start()
+	{
+		SetMusic("Lodge");
+	}
+	
+	/*# region Testing functions	
+
+	[Button()]
+	public void SetLodge()
+	{
+		SetMusic("Lodge");
+	}
+	
+	[Button()]
+	public void SetWinter()
+	{
+		SetMusic("Winter");
+	}
+		
+	[Button()]
+	public void PlayPop()
+	{
+		PlaySound("Pop");
+	}
+	
+			
+	[Button()]
+	public void PlayPop3D()
+	{
+		PlaySound("Pop", Vector3.left);
+	}
+	
+	#endregion*/
+	
+	#region Playing music and sound effects
+
+	/// <summary>
+	/// 2D PlaySound. Creates an audiosource if it doesn't already exist, plays it
+	/// if the audiosource is not already playing.
+	/// </summary>
+	/// <param name="soundName">Name of the sound to play, case sensitive.</param>
+	/// <param name="stereoBlend">Optional argument, -1 is left, +1 is right. 0 by default</param>
+	public void PlaySound(string soundName, float stereoBlend = 0)
+	{
+		AudioSource sound = PlaySoundHelper(soundName);
+		sound.spatialBlend = 0; //2d
+		sound.panStereo = stereoBlend;
+	}
+
+	/// <summary>
+	/// 3D PlaySound overload. Creates an audiosource if it doesn't already exist, plays it
+	/// if the audiosource is not already playing. 
+	/// </summary>
+	/// <param name="soundName">Name of the sound to play, case sensitive.</param>
+	/// <param name="position">World position of the sound to be played.</param>
+	public void PlaySound(string soundName, Vector3 position)
+	{
+		AudioSource sound = PlaySoundHelper(soundName);
+		sound.spatialBlend = 1; //3d
+		sound.transform.position = position;
+	}
+
+	/// <summary>
+	/// Helper function for the play sound functions. Creates an audiosource if it doesn't already exist, plays it
+	/// if the audiosource is not already playing. This function adds the audiosource to the list, but returns a reference
+	/// as, well for convenience.
+	/// </summary>
+	/// <param name="soundName">Name of the sound to play, case sensitive.</param>
+	/// <returns>reference to the audiosource created (or reused)</returns>
+	private AudioSource PlaySoundHelper(string soundName)
+	{
+		if (!_audiosources.ContainsKey(soundName))
+		{
+			GameObject newGO = new GameObject();
+			newGO.transform.parent = transform;
+			newGO.name = soundName;
+			_audiosources.Add(soundName, newGO.AddComponent<AudioSource>());
+			_audiosources[soundName].clip = _audioSettings.sfxDictionary[soundName];
+		}
```

Comment:
I will definitely look for a pooling library as it would be helpful for this type of functionality as well as supporting game-specific development as well.